src/main/java/com/shopping_app/shoppingApp/config/SpringSecurityConfig.java

```diff
@@ -0,0 +1,60 @@
+package com.shopping_app.shoppingApp.config;
+
+import com.shopping_app.shoppingApp.config.JWT.JwtTokenFilter;
+import com.shopping_app.shoppingApp.config.JWT.JwtTokenProvider;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.builders.WebSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.config.http.SessionCreationPolicy;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+
+@Configuration
+@EnableWebSecurity
+@RequiredArgsConstructor(onConstructor = @__({@Autowired}))
+public class SpringSecurityConfig extends WebSecurityConfigurerAdapter {
+
+    private final JwtTokenProvider tokenProvider;
+    private final ProblemAuthenticationEntryPoint authenticationEntryPoint;
+
+    @Bean
+    public PasswordEncoder bCryptPasswordEncoder() {
+        return new BCryptPasswordEncoder();
+    }
+
+    @Bean
+    @Override
+    public AuthenticationManager authenticationManagerBean() throws Exception {
+        return super.authenticationManagerBean();
+    }
+
+    @Override
+    public void configure(WebSecurity web) throws Exception {
+        web.ignoring().antMatchers("/user/login","/user/register");
```

Comment:
can we not have this in the same method that has the http security?

Reply:
Yes we can but, when i added in http security its not working  , so i added this in websecurity layer

Reply:
this functionality alone not changed , reason its not working without websecurity .